CoastalResilienceNetwork/GeositeFramework
# Implement auto-hiding and minimization of legend

* Allow users to minimize the legend when clicking the minimize icon in the legend header.
* If there are no active layers on the map, hide the legend.

Also removes the "legend display" plugin since it is no longer needed.

**Testing instructions**
- When the app loads, the legend should not appear.
- Add a layer, and the legend should appear.
- Click the minimize button in the legend header. The legend should minimize, and the bottom of the legend header should be in line with where the bottom of the legend itself used to be.
- Restore the header. The top of the header should be in the same place as it was when the app was started. 
- Move the legend around and retry the above two steps. 
- Minimize the legend, move it all the way to the top of the window and restore it. Verify that the legend header is not covered by the app title bar.
- Verify that the legend control near the zoom controls is gone.

**Note** I may wait to merge this in until there is a PR for #495, so that the number of times the legend is rendered needlessly can be reduced. 

Connects to #485.
